ROADMAP.md

```diff
@@ -8,124 +7,164 @@ Notes: This document is still in review will may be heavily modified based on st
 https://github.com/libp2p/go-libp2p/issues/1806
 ```
 
+## Table of Contents <!-- omit in toc -->
+
+- [About the Roadmap](#about-the-roadmap)
+  - [Vision](#vision)
+  - [Sections](#sections)
+  - [Done criteria](#done-criteria)
+- [Benchmarking and Testing](#benchmarking-and-testing)
+- [🛣️ Milestones](#️-milestones)
+  - [2022](#2022)
+    - [Early Q4 (October)](#early-q4-october)
+    - [End of Q4 (December)](#end-of-q4-december)
+  - [2023](#2023)
+    - [Early Q1 (January)](#early-q1-january)
+    - [Mid Q1 (February)](#mid-q1-february)
+    - [End of Q1 (March)](#end-of-q1-march)
+  - [Up Next](#up-next)
+- [📖 Appendix](#-appendix)
+  - [A. 📺 Universal Browser Connectivity](#a--universal-browser-connectivity)
+    - [1. WebRTC: Browser to Server](#1-webrtc-browser-to-server)
+    - [2. WebRTC: Browser to Browser](#2-webrtc-browser-to-browser)
+    - [3. WebTransport: Update to new draft versions](#3-webtransport-update-to-new-draft-versions)
+  - [B. ⚡️ Handshakes at the Speed of Light](#b-️-handshakes-at-the-speed-of-light)
+    - [1. Early Muxer Negotiation](#1-early-muxer-negotiation)
+    - [2. Adding security protocol](#2-adding-security-protocol)
+    - [3. 0.5 RTT data optimization](#3-05-rtt-data-optimization)
+  - [C. 🧠 Smart Dialing](#c--smart-dialing)
+    - [1. Happy Eyeballs](#1-happy-eyeballs)
+    - [2. QUIC Blackhole detector](#2-quic-blackhole-detector)
+    - [3. RTT estimation](#3-rtt-estimation)
+  - [D. 📊 Comprehensive Metrics](#d--comprehensive-metrics)
+  - [E. 📢 Judicious Address Advertisements](#e--judicious-address-advertisements)
+- [libp2p Project Roadmap](#libp2p-project-roadmap)
+
 ## About the Roadmap
+
+### Vision
+The tenets of the libp2p team are:
+- Security
+  - We strive to ensure that libp2p is reliable & secure for users & applications. We seek to minimize vulnerabilities and act quickly to address issues.
+- Stability
+  - We do not break deployed features and maintain cross version compatibility across libp2p releases.
+- Performance
+  - We aim to make libp2p as performant as possible and strive to ensure comparable performance to widely used protocols on the Internet. We avoid regressions and performance degradations in new releases.
```

Comment:
Idea: move this to specs ROADMAP and link to it there so we have this info in a canonical place.